frontend/src/lib/k8s/api/v2/webSocket.ts

```diff
@@ -167,3 +169,93 @@ export function useWebSocket<T>({
     };
   }, [enabled, url, protocols, type, cluster]);
 }
+
+export type WebSocketConnectionRequest<T> = {
+  cluster: string;
+  url: string;
+  onMessage: (data: T) => void;
+};
+
+/**
+ * Creates or joins mutiple existing WebSocket connections
+ *
+ * @param url - endpoint URL
+ * @param options - WebSocket options
+ */
+export function useWebSockets<T>({
+  connections,
+  enabled = true,
+  protocols,
+  type = 'json',
+}: {
+  enabled?: boolean;
+  connections: Array<WebSocketConnectionRequest<T>>;
+  /**
+   * Any additional protocols to include in WebSocket connection
+   */
+  protocols?: string | string[];
+  /**
+   * Type of websocket data
+   */
+  type?: 'json' | 'binary';
+}) {
+  useEffect(() => {
+    if (!enabled) return;
+
+    let isCurrent = true;
+
+    function connect({ cluster, url, onMessage }: WebSocketConnectionRequest<T>) {
+      const connectionKey = cluster + url;
+
+      if (sockets.has(connectionKey)) return () => {};
+
+      // Add new listener for this URL
+      listeners.set(connectionKey, [...(listeners.get(connectionKey) ?? []), onMessage]);
+
+      // Mark socket as pending, so we don't open more than one
+      sockets.set(connectionKey, 'pending');
+
+      let ws: WebSocket | undefined;
+      openWebSocket(url, { protocols, type, cluster, onMessage }).then(socket => {
+        ws = socket;
+
+        // Hook was unmounted while it was connecting to WebSocket
+        // so we close the socket and clean up
+        if (!isCurrent) {
+          ws.close();
+          sockets.delete(connectionKey);
+          return;
+        }
+
+        sockets.set(connectionKey, ws);
+      });
```

Comment:
what happens if this returns an error here for the websocket?

Reply:
added a catch clause